src/App.tsx

```diff
@@ -14,11 +16,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const lengthFromServer: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const App: React.FC = () => {
+  const [state, setState] = useState({
+    goods: [...goodsFromServer],
+    isVisible: false,
+    length: 1,
+  });
```

Comment:
better create separate states. Now if you want to update one value you need to add all values on setState

```suggestion
  const [goods, setGoods] = useState(...);
  const [isVisible,] = useState();
  ...
```